pkgs/servers/sql/cockroachdb/default.nix

```diff
@@ -1,35 +1,40 @@
-{ stdenv, buildGoPackage, fetchurl, cmake, xz, which, autoconf }:
+{ stdenv, buildGoPackage, fetchurl, go, cmake, which, autoconf, libedit }:
 
-buildGoPackage rec {
+stdenv.mkDerivation rec {
```

Comment:
Please stick to `buildGoPackage` even if the build phase is overwritten. Otherwise cockroach will have runtime references to the go compiler, which should be avoided.

Reply:
Ah, OK. In that case #38662 should be preferred, then.

Reply:
Note that it's not just go - only ncurses/libedit are needed at runtime; all the other deps (cmake, autoconf, etc) are build-time-only. Ideally those would also be filtered out the same way that buildGoPackage does for the go compiler.

Reply:
This is detected by nix. The problem with go is that string references in binaries are embedded and therefore create a runtime dependency on the compiler.